### ISSUE TITLE
Initial platform download failed

### ISSUE THREAD

**Issue by ajnasaboobacker** [Sharrnah/whispering-ui#23]
![image](https://github.com/Sharrnah/whispering-ui/assets/47502431/01084a54-e975-46ec-b04c-a79c338d9dfe)

The initial platform download is not working for me. 

From the log i can see the below:

Error: Could not start audioWhisper

**Comment by Sharrnah** [comment]
Do you use adguard?

Can you try disabling it temporarily?

My guess is that the Adguard certificate has some issue. You could also try to reinstall the certificate of adguard.
![grafik](https://github.com/Sharrnah/whispering-ui/assets/55756126/f7fc0088-a258-4120-bfce-552471b1a70e)

**Comment by ajnasaboobacker** [comment]
It worked after connecting a VPN for downloading..

But now the issue is i am getting an error "MiniAudio: Permission denied"

**Comment by Sharrnah** [comment]
That sounds like an issue with getting the audio device.
Do you get that error already when Audio Devices are set to "Default" ?

Possible reasons / solutions:
- Your Antivirus / Firewall is blocking Apps from accessing the selected microphone. Try to give permissions to the Application in your firewall.
- You selected the wrong Audio Device for either Input or Output.
   - Try a different audio device and see if the error does not appear for the other device.
- Try to change to the MME Audio API. (This one does not support Loopback devices though, so depending on your usecase you might need VoiceMeeter)

**Comment by ajnasaboobacker** [comment]
Yes, This is showing the same error when devices are set to "Default"

I validated the firewall and nothing seems to be blocking.

It is same for all the devices listed, same error.

![image](https://github.com/Sharrnah/whispering-ui/assets/47502431/ec0b6071-fc2b-4a08-9db8-34f1672a1a6e)

**Comment by ajnasaboobacker** [comment]
attaching the log file here.

[log.txt](https://github.com/user-attachments/files/15962043/log.txt)

**Comment by Sharrnah** [comment]
Have you tried switching to the MME Audio API?
![grafik](https://github.com/Sharrnah/whispering-ui/assets/55756126/eb756dbf-95a9-46f7-8226-e1285d8da8d2)

According to your Log you first get the error "Invalid sample rate" where MME is way less specific since in my testing provides already the correct sample_rate to the application. Whereas in WASAPI i have to resample myself.

(Still odd that it does not just resample but completely crashes, even after using the sample_rate that the audio device claims to use. 🤔 )

Could you also maybe provide a list of your audio devices? Maybe it is some form of naming conflict.

**Comment by ajnasaboobacker** [comment]
This is what i see when i select MME.
![image](https://github.com/Sharrnah/whispering-ui/assets/47502431/e69c49c7-3098-453f-acfa-279cfdf36996)

List of audio devices:

![image](https://github.com/Sharrnah/whispering-ui/assets/47502431/ec54bd29-2d81-41bc-b551-9f9640f6e4bf)

**Comment by Sharrnah** [comment]
Thank you.

Except the MIDI devices i don't see anything out of the ordinary (compared to my list).
I would not expect that MIDI devices are considered by the audio library though.

But the error when switching to MME is a bit different and might tell that it is not able to use the audio output device.

- Which of the shown audio devices is your windows default?
- Do you maybe have something running in the background which already uses the audio device?
- Can you try and manually select your correct audio output device in the Whispering Tiger Profile while using the MME Audio API?
- Can you check if you allowed apps to access your microphone in the windows settings?
   ![grafik](https://github.com/Sharrnah/whispering-ui/assets/55756126/7b96e861-dfcd-4e2c-930c-e909f99d306b)


And sorry about trying these things as i am just guessing currently what might be the issue here.
